sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/json/JSONOptions.scala

```diff
@@ -55,7 +55,7 @@ class JSONOptions(
   private val maxStringLen: Int = parameters
     .get("maxStringLen")
     .map(_.toInt)
-    .getOrElse(StreamReadConstraints.DEFAULT_MAX_STRING_LEN)
+    .getOrElse(SQLConf.get.getConf(SQLConf.JSON_MAX_STRING_LENGTH))
```

Comment:
can we follow `ParquetOptions` and pass a `SQLConf` instance to construct `JSONOptions`?